src/neighbors.cc

```diff
@@ -165,14 +165,18 @@ peerAllowedToUse(const CachePeer * p, HttpRequest * request)
         return true;
 
     ACLFilledChecklist checklist(p->access, request, NULL);
-//    checklist.al = ps->al;
+    checklist.al = ps->al;
+    if (ps->al && ps->al->reply) {
+        checklist.reply = ps->al->reply;
```

Comment:
Do we have to copy this poiinter because v4 Checklist cannot get `reply` from `al`? If yes, would it be overall better to fix that v4 shortcoming instead?

Reply:
> Do you suggest that ...

Maybe. I am concerned that the code adds information (`.reply`) that is already available (`.al->reply`). That does not look right. I do _not_ know what the correct solution to that problem is -- I have not studied it.

> I am afraid this would require too many out-of-scope changes, I would rather do it in a separate PR.

I am OK with solving this problem in a separate, dedicated PR. We should probably mention that plan in the official PR discussion (when this PR becomes official and, at least, in the official PR 447 that suffers from the same ailment).